File: aqvp-admin-service/src/main/java/com/aqvp/platform/admin/service/NotificationService.java
package com.aqvp.platform.admin.service;

import com.aqvp.platform.admin.domain.Notification;
import com.aqvp.platform.admin.domain.NotificationChannel;

public interface NotificationService {

    Notification sendNotification(String templateCode,
                                NotificationChannel channel,
                                String recipient,
                                String subject,
                                String message);
}

File: frontend/aqvp-web/src/features/notifications/services/notificationService.ts
import { adminApi } from '@/config/axios';
import { API_ENDPOINTS } from '@/constants/api';

export type NotificationChannel = 'EMAIL' | 'SMS';
export type NotificationStatus = 'PENDING' | 'SENT' | 'FAILED' | 'RETRYING';

export interface SendNotificationRequest {
  templateCode: string;
  channel: NotificationChannel;
  recipient: string;
  subject?: string;
  message: string;
}

export interface Notification extends SendNotificationRequest {
  id: string;
  status: NotificationStatus;
  attempts?: number;
  sentAt?: string;
  errorMessage?: string;
  providerReference?: string;
  providerResponse?: string;
}

export const notificationService = {
  send: async (payload: SendNotificationRequest): Promise<Notification> => {
    const response = await adminApi.post<Notification>(API_ENDPOINTS.ADMIN.NOTIFICATIONS, payload);
    return response.data;
  },
};
